test/Domain/LeanCode.DomainModels.Tests/Model/TimestampTzTests.cs

```diff
@@ -0,0 +1,171 @@
+    {
+        var timestampTz = new TimestampTz(DateTime.UtcNow, TimeZoneInfo.Utc);
+
+        Assert.Equal("Etc/UTC", timestampTz.TimeZoneId);
+        Assert.Equal(TimeZoneInfo.FindSystemTimeZoneById("Etc/UTC"), timestampTz.TimeZoneInfo);
+    }
+
+    [Fact]
+    public void Windows_time_zone_ids_are_converted_to_IANA_ids()
+    {
+        var timestampTz = new TimestampTz(DateTime.UtcNow, "Central European Standard Time");
+
+        Assert.Equal("Europe/Warsaw", timestampTz.TimeZoneId);
+        Assert.Equal(TimeZoneInfo.FindSystemTimeZoneById("Europe/Warsaw"), timestampTz.TimeZoneInfo);
+    }
+
+    [Fact]
+    public void Windows_time_zone_infos_are_converted_to_IANA_ids()
+    {
+        var timestampTz = new TimestampTz(
+            DateTime.UtcNow,
+            TimeZoneInfo.FindSystemTimeZoneById("Central European Standard Time")
+        );
+
+        Assert.Equal("Europe/Warsaw", timestampTz.TimeZoneId);
+        Assert.Equal(TimeZoneInfo.FindSystemTimeZoneById("Europe/Warsaw"), timestampTz.TimeZoneInfo);
+    }
+
+    [Theory]
+    [InlineData("")]
+    [InlineData("invalid")]
+    public void Invalid_time_zone_ids_throw_exception(string timeZoneId)
+    {
+        Assert.Throws<TimeZoneNotFoundException>(() => new TimestampTz(DateTime.UtcNow, timeZoneId));
+    }
+
+    [Fact]
+    public void Non_UTC_DateTime_timestamps_throw_exception()
+    {
+        Assert.Throws<ArgumentException>(() => new TimestampTz(DateTime.Now, "Etc/UTC"));
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void Non_UTC_DateTimeOffset_timestamps_are_converted_to_UTC(bool useTimeZoneInfo)
+    {
+        var now = DateTimeOffset.Now;
+
+        var timestampTz = useTimeZoneInfo
+            ? new TimestampTz(now, TimeZoneInfo.FindSystemTimeZoneById("Etc/UTC"))
+            : new TimestampTz(now, "Etc/UTC");
+
+        Assert.Equal(now.UtcDateTime, timestampTz.UtcTimestamp);
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void Local_timestamps_are_correctly_calculated(bool useTimeZoneInfo)
+    {
+        var now = DateTimeOffset.UtcNow;
+        var nowInWarsaw = TimeZoneInfo.ConvertTimeBySystemTimeZoneId(now, "Europe/Warsaw");
+
+        var timestampTz = useTimeZoneInfo
+            ? new TimestampTz(now, TimeZoneInfo.FindSystemTimeZoneById("Europe/Warsaw"))
+            : new TimestampTz(now, "Europe/Warsaw");
+
+        Assert.Equal(nowInWarsaw, timestampTz.LocalTimestampWithOffset);
+        Assert.Equal(nowInWarsaw.DateTime, timestampTz.LocalTimestampWithoutOffset);
+    }
+
+    [Fact]
+    public void Deconstruction_roundtrips()
+    {
+        var source = new TimestampTz(DateTime.UtcNow, "Europe/Warsaw");
+        var (timestamp, timeZoneId) = source;
+        var reconstructed = new TimestampTz(timestamp, timeZoneId);
+
+        Assert.Equal(source, reconstructed);
+    }
+
+    [Fact]
+    public void Json_serialization_roundtrips()
```

Comment:
Let's split this test into two - one that check if the serialization format is indeed like we expect, and the other that checks the roundtripping.